test/IISTestSite/StartupLargeResponseBody.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace IISTestSite
+{
+    public class StartupLargeResponseBody
+    {
+        public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
+        {
+            app.Run(async context =>
+            {
+                if (context.Request.Path.Equals("/LargeResponseBody"))
+                {
+                    // TODO is there a better way to remove the ? from the query string
+                    if (int.TryParse(context.Request.QueryString.Value.Substring(1), out var length))
```

Comment:
What is this trying to do?

Reply:
Send a request with the length of the responsebody.

Reply:
Why not just read the specific query string key then?